src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -95,6 +99,8 @@
     "TIMESTAMPFORMAT": "TIMESTAMP_FORMAT",
 }
 
+MAX_RETRY_TIME = 3
```

Comment:
the imports are causing trouble in test pipelines:
```python
from pyodbc import Connection
import pandas as pd
```

1. we should not have pyodbc have our dependency, it's controlled by users, you can check if python has its own dbapi connection type hint or add by yourself
2. pandas is optional, you can check how other places handle pandas optional import